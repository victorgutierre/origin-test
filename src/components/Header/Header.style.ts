import styled from 'styled-components';

export const HeaderContainer = styled.header`
  display: block;
  width: 100%;
  padding: 1rem 3rem;
  background: #fff;
`;

HeaderContainer.displayName = 'header';
