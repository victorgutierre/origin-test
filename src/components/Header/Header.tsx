import React from 'react';
import { HeaderContainer } from './Header.style';
interface Props {
  logo: string;
}

export const Header: React.FC<Props> = ({ logo }) => (
  <HeaderContainer data-testid="header">
    <img src={logo} width="100" height="50" data-testid="header-logo" />
  </HeaderContainer>
);
