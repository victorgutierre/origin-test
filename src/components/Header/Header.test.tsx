import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('<Header />', () => {
  const logo = './icons/logo.svg';
  const component = <Header logo={logo} />;

  it('Should render Header component', () => {
    const wrapper = shallow(component);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Header component elements render', () => {
    const wrapper = shallow(component);
    expect(wrapper.find('header')).toHaveLength(1);
  });

  it('Contains a tag image inside header', () => {
    const wrapper = shallow(component);
    const img = <img src={logo} />;
    expect(wrapper.containsMatchingElement(img)).toBe(true);
  });

  it('Img src should be logo.svg', () => {
    const wrapper = shallow(component);
    const img = wrapper.find('img');
    expect(img.props().src).toBe('./icons/logo.svg');
  });
});
