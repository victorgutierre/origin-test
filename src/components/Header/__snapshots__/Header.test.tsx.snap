// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render Header component 1`] = `
<header>
  <img
    height="50"
    src="./icons/logo.svg"
    width="100"
  />
</header>
`;
